### PR TITLE
Fix in app reviews crash

### PR DESCRIPTION
**Pull request by irfano**
There are a few crash reports that occurred when logging the crash exception. 
- I created an extension function to log the exception from a single place.
- If the exception can't be cast to `ReviewException`, it will log the `Exception`'s message.

-----

## To Test:

This change cannot be tested using the .apk generated in this PR. The .apk should be uploaded to [internal app sharing](https://developer.android.com/guide/playcore/in-app-review/test#internal-app-sharing). Even if we do so, reproducing the exception is not easy.

I am confident that the change is safe. Please review it and let me know if you believe it requires specific testing.

-----

## Regression Notes

1. Potential unintended areas of impact

    - None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Nothing

3. What automated tests I added (or what prevented me from doing so)

    - Nothing. I don't believe it's worthwhile to mock `ReviewException` and add a unit test. The `Task<ReviewInfo>.logException()` logic is quite basic.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-Android/milestone/275">24.1 ❄️</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/java/org/wordpress/android/ui/utils/InAppReviewsUtils.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.wordpress.android.ui.utils
+
+import com.google.android.gms.tasks.Task
+import com.google.android.play.core.review.ReviewException
+import com.google.android.play.core.review.ReviewInfo
+import com.google.android.play.core.review.model.ReviewErrorCode
+import org.wordpress.android.util.AppLog
+
+fun Task<ReviewInfo>.logException() {
```
Comment (notandyvee) [inline WordPress/src/main/java/org/wordpress/android/ui/utils/InAppReviewsUtils.kt]: Could you actually move `InAppReviewsUtils` to:

    WordPress/src/main/java/org/wordpress/android/util/extensions/InAppReviewsUtils.kt

Would keep all our extension functions nice and tidy.

Comment (irfano) [reply WordPress/src/main/java/org/wordpress/android/ui/utils/InAppReviewsUtils.kt]: Good point! Done: 7cbbb26d3a833f40c2db6ef8d8d5b20ac4122134

**Review by notandyvee — APPROVED** [review]
I agree with Irfan. Low risk. LGTM!

**Pull request merged**